pgxn/neon/communicator.c

```diff
@@ -2568,6 +2622,14 @@ communicator_processinterrupts(void)
 		if (!readpage_reentrant_guard && readahead_getpage_pull_timeout_ms > 0)
 			communicator_prefetch_pump_state();
 
+#if PG_MAJORVERSION_NUM <= 14
+		/*
+		 * PG14: Setting a repeating timeout is not possible, so we signal here
+		 * that the timeout has already been reset, and by telling the system
+		 * that system will re-schedule it later if we need to.
+		 */
+		timeout_set = false;
+#endif
```

Comment:
Why was this moved?

Reply:
Setting timeout in signal handler can cause race condition which in turn can cause assertion failure in communicator.c:2596:
```
			Assert(timeout_set);
```